### ISSUE TITLE
'semi' rule complains about VariableDeclaration in for-loop initializer

### ISSUE THREAD

**Issue by mamacdon** [eslint/eslint#209]
The `semi` rule warns on a VariableDeclaration that appears in a for-loop initializer. For example, I expected this code to pass:

``` js
for (var i;;){}
```

But instead it generates:
`for.js: line 1, col 5, Warning - Missing semicolon.`
